### PR TITLE
feat: add your projects (with roles) to personal dashboard api

### PR DESCRIPTION
**Pull request by thomasheartman**
This PR adds some of the necessary project data to the personal dashboard API: project names and ids, and the roles that the user has in each of these projects.

I have not added project owners yet, as that would increase the complexity a bit and I'd rather focus on that in a separate PR.

I have also not added projects you are part of through a group, though I have added a placeholder test for that. I will address this in a follow-up.

### CODE REVIEW

File: src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts [hunk 1]
```diff
@@ -61,3 +62,84 @@ test('should return personal dashboard with own flags and favorited flags', asyn
...
+            name,
+        },
+        user,
+        auditUser,
+    );
+    return project;
+};
+
+test('should return personal dashboard with membered projects', async () => {
+    // create project A with user 1
```
Comment (kwasniew) [inline src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts:83]: those comments don't tell me anything I can't learn from the code itself

File: src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts [hunk 2]
```diff
@@ -61,3 +62,84 @@ test('should return personal dashboard with own flags and favorited flags', asyn
...
+    await createProject('Project B', user1);
+
+    // create project C with user 2
+    const { body: user2 } = await loginUser('user2@test.com');
+    const projectC = await createProject('Project C', user2);
+
+    // Add user 2 as a member of project A
+    await app.services.projectService.addAccess(
+        projectA.id,
+        [5],
```
Comment (kwasniew) [inline src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts:96]: what is 5?

**Review by kwasniew — APPROVED** [review]
Nice! Minor comments

File: src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts (hunk 1 above)
Comment (thomasheartman) [reply src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts:83]: fair; I'll remove them

File: src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts (hunk 2 above)
Comment (thomasheartman) [reply src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts:96]: That is the ID for project members. I guess I could add a comment here; will fix

**Pull request merged**